internal/processing/processors/migration/annotation.go

```diff
@@ -0,0 +1,41 @@
+package migration
+
+import (
+	"context"
+	gatewayv2alpha1 "github.com/kyma-project/api-gateway/apis/gateway/v2alpha1"
+	"github.com/kyma-project/api-gateway/internal/processing"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type annotationProcessor struct {
+	config processing.ReconciliationConfig
+	api    *gatewayv2alpha1.APIRule
+}
+
+const (
+	annotationName = "gateway.kyma-project.io/migration-step"
+
+	applyIstioAuthorizationMigrationStep MigrationStep = "apply-istio-authorization"
+	switchVsToService                    MigrationStep = "vs-switch-to-service"
+	removeOryRule                        MigrationStep = "remove-ory-rule"
+	finished                             MigrationStep = "migration-finished"
```

Comment:
Do we use finished at all since we are removing the annotation if the nextMigrationStep is `remove-ory-rule`.